src/browser/api_protocol/api_handlers/mesh_middleware.ts

```diff
@@ -128,6 +134,85 @@ function sendMessageMiddleware(msg: MessagePackage, next: () => void) {
     }
 }
 
+const handleExternalWindow = async (identity: Identity, toGroup: Identity): Promise<string|void> => {
+    const { uuid, name } = toGroup;
+    if (coreState.getWindowByUuidName(identity.uuid, name)) {
+        // External window already created and registered
+        return 'registered';
+    }
+    if (!uuid || isLocalUuid(uuid)) {
+        return;
+    }
+    try {
+        const getHwndMessage = {
+            action: 'get-window-native-id',
+            payload: {
+                uuid,
+                name
+            }
+        };
+        const id = await connectionManager.resolveIdentity({uuid});
+        const hwnd = await id.runtime.fin.System.executeOnRemote(identity, getHwndMessage);
+
+        const childWindowOptions = {
+            uuid: identity.uuid,
+            name,
+            hwnd: hwnd.data,
```

Comment:
So to clarify some terminology, an "external window" is a window that wasn't created by OpenFin. This could something like a Google Chrome window or an Excel window.

Now if in a multi-runtime setup you're trying to `joinGroup` with another OpenFin window, you shouldn't be creating a new BrowserWindow. What happens as a result is that now there are two BrowserWindows per native window, which can cause some weird stuff to happen, as you pointed out.

What you should be doing to refer to another OpenFin window is using `wrap`. And from there you can use `getNativeId` to get the `hwnd` to use in Window Transactions for group moving.